react/table/src/components/common/ConfirmationDialogTrigger.js

```diff
@@ -0,0 +1,83 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+
+export default class ConfirmationDialogTrigger extends PureComponent {
+  static CONFIRM = 'CONFIRM';
+
+  static DISCARD = 'DISCARD';
+
+  constructor(props) {
+    super(props);
+    this.handleOpen = this.handleOpen.bind(this);
+    this.state = { open: false };
+  }
+
+  handleOpen() {
+    this.setState(() => ({
+      open: true,
+    }));
+  }
+
+  handleClose(action) {
+    const { onCloseDialog } = this.props;
+    this.setState(() => ({
+      open: false,
+    }));
+    onCloseDialog(action);
+  }
+
+  render() {
+    const { open } = this.state;
+    const {
+      dialog: { title, description, confirmLabel, discardLabel },
+      Renderer,
+    } = this.props;
+    return (
+      <div>
+        <Renderer onClick={this.handleOpen} />
+        <Dialog
+          open={open}
+          onClose={this.handleClose}
+          aria-labelledby="confirmation-dialog-title"
+          aria-describedby="confirmation-dialog-description"
+        >
+          <DialogTitle id="confirmation-dialog-title">{title}</DialogTitle>
+          <DialogContent>
+            <DialogContentText id="confirmation-dialog-description">
+              {description}
+            </DialogContentText>
+          </DialogContent>
+          <DialogActions>
+            <Button onClick={() => this.handleClose(ConfirmationDialogTrigger.DISCARD)} autoFocus>
```

Comment:
They used to suggest avoiding creating functions inside render

Reply:
correct, thank! updated